rust-runtime/aws-smithy-experimental/src/hyper_1_0.rs

```diff
@@ -0,0 +1,1223 @@
+    use crate::hyper_1_0::build_connector::make_tls;
+    use crate::hyper_1_0::{CryptoMode, Inner};
+
+    #[cfg(feature = "crypto-ring")]
+    pub(crate) static HTTPS_NATIVE_ROOTS_RING: once_cell::sync::Lazy<
+        HttpsConnector<HttpConnector>,
+    > = once_cell::sync::Lazy::new(|| make_tls(GaiResolver::new(), CryptoMode::Ring.provider()));
+
+    #[cfg(feature = "crypto-aws-lc")]
+    pub(crate) static HTTPS_NATIVE_ROOTS_AWS_LC: once_cell::sync::Lazy<
+        HttpsConnector<HttpConnector>,
+    > = once_cell::sync::Lazy::new(|| make_tls(GaiResolver::new(), CryptoMode::AwsLc.provider()));
+
+    pub(super) fn cached_https(mode: Inner) -> hyper_rustls::HttpsConnector<HttpConnector> {
+        match mode {
+            #[cfg(feature = "crypto-ring")]
+            Inner::Standard(CryptoMode::Ring) => HTTPS_NATIVE_ROOTS_RING.clone(),
+            #[cfg(feature = "crypto-aws-lc")]
+            Inner::Standard(CryptoMode::AwsLc) => HTTPS_NATIVE_ROOTS_AWS_LC.clone(),
+            #[allow(unreachable_patterns)]
+            Inner::Standard(_) => unreachable!("unexpected mode"),
+            Inner::Custom(provider) => make_tls(GaiResolver::new(), provider),
+        }
+    }
+}
+
+mod build_connector {
+    use std::sync::Arc;
+
+    use client::connect::HttpConnector;
+    use hyper_rustls::HttpsConnector;
+    use hyper_util::client::legacy as client;
+    use rustls::crypto::CryptoProvider;
+
+    use aws_smithy_runtime_api::client::dns::ResolveDns;
+
+    use crate::hyper_1_0::{HyperUtilResolver, Inner};
+
+    fn restrict_ciphers(base: CryptoProvider) -> CryptoProvider {
+        let suites = &[
+            rustls::CipherSuite::TLS13_AES_256_GCM_SHA384,
+            rustls::CipherSuite::TLS13_AES_128_GCM_SHA256,
+            // TLS1.2 suites
+            rustls::CipherSuite::TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+            rustls::CipherSuite::TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+            rustls::CipherSuite::TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+            rustls::CipherSuite::TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+            rustls::CipherSuite::TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+        ];
+        let supported_suites = suites
+            .iter()
+            .flat_map(|suite| {
+                base.cipher_suites
+                    .iter()
+                    .find(|s| &s.suite() == suite)
+                    .cloned()
+            })
+            .collect::<Vec<_>>();
+        CryptoProvider {
+            cipher_suites: supported_suites,
+            ..base
+        }
+    }
+
+    pub(crate) fn make_tls<R>(
+        resolver: R,
+        crypto_provider: CryptoProvider,
+    ) -> HttpsConnector<HttpConnector<R>> {
+        use hyper_rustls::ConfigBuilderExt;
+        hyper_rustls::HttpsConnectorBuilder::new()
+               .with_tls_config(
+                rustls::ClientConfig::builder_with_provider(Arc::new(restrict_ciphers(crypto_provider)))
+                    //.with_safe_default_kx_groups()
```

Comment:
?

Reply:
ah they removed that — it is only safe now.